src/imu_control_parameters.cpp

```diff
@@ -0,0 +1,524 @@
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#include "imu_ros2/imu_control_parameters.h"
+#include <thread>
+
+
+ImuControlParameters::ImuControlParameters(std::shared_ptr<rclcpp::Node>& node)
+{
+    init(node);
+}
+
+ImuControlParameters::~ImuControlParameters()
+{
+
+}
+
+void ImuControlParameters::init(std::shared_ptr<rclcpp::Node> &node)
+{
+    m_node = node;
+
+    m_operation_mode = 0;
+    m_filter_size = 0;
+    m_burst_size_selection = 0;
+    m_burst_data_selection = 0;
+    m_linear_acceleration_compensation = 1;
+    m_point_of_percussion_alignment = 1;
+    m_internal_sensor_bandwidth = 0;
+    m_sync_mode_select = 3;
+    m_sync_polarity = 0;
+    m_data_ready_polarity = 1;
+    m_sync_signal_scale = 2000;
+    m_decimation_filter = 0;
+    m_accel_calibbias_x = 0;
+    m_accel_calibbias_y = 0;
+    m_accel_calibbias_z = 0;
+    m_anglvel_calibbias_x = 0;
+    m_anglvel_calibbias_y = 0;
+    m_anglvel_calibbias_z = 0;
+    m_sampling_frequency = 2000;
+}
+
+void ImuControlParameters::run()
+{
+    std::thread::id this_id = std::this_thread::get_id();
+    std::cout << "thread parameter " << this_id << " started...\n";
+    RCLCPP_INFO(rclcpp::get_logger("rclcpp_parameter"), "startThread: '%d'", this_id);
+
+    rclcpp::WallRate loopRate(0.1);
+
+    bool hasError = false;
+
+    while (rclcpp::ok())
+    {
+        if(hasError == true)
+        {
+            RCLCPP_INFO(rclcpp::get_logger("rclcpp_set_parameter"), "Error: an libiio parameter value is incorect and was not set");
+
+            rclcpp::Parameter sparam{rclcpp::Parameter("operation_mode", 2)};
+            m_node->set_parameter(sparam);
+            break;
+        }
+
+        int32_t operation_mode = m_node->get_parameter("operation_mode").get_parameter_value().get<int32_t>();
+
+        if( operation_mode == 0)
+        {
+            RCLCPP_INFO(rclcpp::get_logger("rclcpp_set_parameter"), "operation_mode = %d, you can change iio ros paramenters", operation_mode);
+            m_operation_mode = operation_mode;
+        }
+        else
+            if( operation_mode == 1)
+            {
+                RCLCPP_INFO(rclcpp::get_logger("rclcpp_set_parameter"), "operation_mode = %d, you can not change iio ros paramenters", operation_mode);
+
+                loopRate.sleep();
+                continue;
+            }
+            else
+            {
+                RCLCPP_INFO(rclcpp::get_logger("rclcpp_set_parameter"), "operation_mode = %d, invalid value, set 0 or 1", operation_mode);
+
+                loopRate.sleep();
+                continue;
+            }
+
+        int32_t filter_size = m_node->get_parameter("filter_size").get_parameter_value().get<int32_t>();
```

Comment:
there are a lot of code lines here. lines 104-519, and if we add more parameters even more will be added.
there are better ways to do this.
one example would be to create an array of  structures for the parameter name and pointer to the APIs to be called and a function which iterates over the array and basically does the same thing without the extra lines of code, 

The function could look something like:

```
void check_parameter(uint32_t parameter_index, int32_t expected_param_value)
{
int32_t param_value = m_node->get_parameter(array[parameter_index].name).get_parameter_value().get<int32_t>();
if (param_value != expected_param_value)
            int ret = array[parameter_index].function(param_value);
// etc
}
```

Reply:
I resolved this with pointer on method

Reply:
I resolved this with pointer on method